### PR TITLE
Begin Migration for Professor Tags View

### PR DESCRIPTION
**Pull request by Glitched**
**Note:** Review after the AddQuestion PR because this builds on that one
The goal of this PR is to fetch necessary data from the server for ProfessorTagsView

### Inside This PR 

* Lots of refactoring, damn this got overly complex.
* Fetch data from Firebase
* First attempts at a wrapper hook component to pass in data without forcing a component rewrite.

### Test Plan 
Manual QA

### Notes 

Again we'll need cloud functions for mutations.

### Breaking Changes 

- [ ] Database schema change (anything that changes Postgres)
- [x] I updated existing types in `/src/components/types/`
- [x] Other change that could cause problems (Detailed in notes)

### Checklist

- [ ] My changes requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

### CODE REVIEW

**Comment by Glitched** [comment]
@BobbyVillaluz Not sure why the build is failing - did something change in the build process?

File: client/src/components/includes/ProfessorAddNew.tsx [hunk 1]
```diff
@@ -1,74 +1,51 @@
...
-            <div className="ProfessorAddNew">
-                <div className={'Add ' + !this.state.editVisible}>
-                    <button className="NewOHButton" onClick={() => this.toggleEdit(true)}>
+const ProfessorAddNew = (props: {
+    courseId: string
+    taOptions?: DropdownItemProps[]
+}) => {
+    const [editVisible, setEditVisible] = useState(false);
+
+    var text = props.taOptions ? 'Add New Office Hour' : 'Add New Assignment';
```
Comment (BobbyVillaluz) [inline client/src/components/includes/ProfessorAddNew.tsx]: Could we switch this to a `let`?

File: client/src/components/includes/ProfessorTagInfo.tsx [hunk 2]
```diff
@@ -88,135 +62,39 @@ class ProfessorTagInfo extends React.Component {
 
     handleActiveChange = (active: boolean): void => {
         var newState = Object.assign({}, this.state.tag);
```
Comment (BobbyVillaluz) [inline client/src/components/includes/ProfessorTagInfo.tsx]: Could we change this to a `let`?

File: client/src/components/includes/ProfessorTagInfo.tsx [hunk 3]
```diff
@@ -88,135 +62,39 @@ class ProfessorTagInfo extends React.Component {
...
+        console.log('RYAN_TODO add child tag: ', newTag);
     }
 
     handleNewTagEnter = (): void => {
         if (this.state.newTagText.length === 0) {
             return;
         }
-        var newTag: AppTag = {
-            activated: true,
+        var newTag: FireTag = {
```
Comment (BobbyVillaluz) [inline client/src/components/includes/ProfessorTagInfo.tsx]: `let`?

**Pull request merged**